File: src/code/java/_121.java
package code.java;

public class _121 {
    public int maxProfit(int[] prices) {
        int max_profit = 0, min_price = Integer.MAX_VALUE;
        for (int price : prices) {
            min_price = Math.min(min_price, price);
            max_profit = Math.max(max_profit, price - min_price);
        }
        return max_profit;
    }
}

File: src/code/js/_121.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    var max_profit = 0, min_price = prices[0];
    for (var i = 1; i < prices.length; i++) {
        if (prices[i] > min_price) {
            max_profit = Math.max(max_profit, prices[i] - min_price);
        } else {
            min_price = prices[i];
        }
    }
    return max_profit;
};